### DISCUSSION TITLE
SetOption74 does not work for my esp32-c3

### DISCUSSION THREAD

**Discussion by deep414** [arendst/Tasmota#21090] [Q&A] [answered]
I have luatos.com/t/esp32c3 device. I use it to collect the temperature of my boiler. I use DS18B20 sensor attached to GPIO4. It works OK with external pullup resistor. 
In https://tasmota.github.io/docs/DS18x20/#wiring I see that I can use SetOption74 to enable/disable internal pullup resistor when using a single DS18x20 sensor. But when I set SetOption74 to 1 and remove external pullup resister I do not see the temperature on main menu screen. If I add a external pullup resistor temperature appears on the screen.
So, the question: what did I do wrong? Is it possible to use internal pullup resistor on ESP32-c3?
wbr , Alexei.

**Comment by sfromis** [comment]
`SetOption74` only works for ESP8266, with ESP32 variants you need external pullup. Different DS18x20 drivers are used for the architectures, and the ESP32 one does not do anything for pullup.

**Reply by deep414** [reply, marked as answer]
Clear. No more questions. 
Thanks.
